File: MyCallBagEyeNeurology/app/src/main/java/com/opticpro/suite/ui/theme/Theme.kt
package com.opticpro.suite.ui.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color

private val LightColors = lightColorScheme(
    primary = LogoBlack,
    onPrimary = LogoWhite,
    secondary = LogoGray,
    background = LogoWhite,
    onBackground = LogoBlack,
    surface = LogoWhite,
    onSurface = LogoBlack,
    surfaceVariant = Color(0xFFE8E8E8),
)

private val DarkColors = darkColorScheme(
    primary = LogoWhite,
    onPrimary = LogoBlack,
    secondary = LogoGray,
    background = LogoBlack,
    onBackground = LogoWhite,
    surface = LogoSurface,
    onSurface = LogoWhite,
    surfaceVariant = LogoSurfaceVariant,
    onSurfaceVariant = LogoGray,
)

// Stile monocromatico bianco su nero ispirato al logo dell'app: tema scuro di default.
@Composable
fun OpticProSuiteTheme(darkTheme: Boolean = true, content: @Composable () -> Unit) {
    val colors = if (darkTheme) DarkColors else LightColors
    MaterialTheme(colorScheme = colors, content = content)
}

File: MyCallBagEyeNeurologyIOS/Optools/Theme/Theme.swift
// © Roberto Di Flumeri
import SwiftUI

/// Palette monocromatica ispirata al logo (anelli bianchi su sfondo nero).
enum AppColor {
    static let logoBlack = Color.black
    static let logoSurface = Color(red: 0x16 / 255, green: 0x16 / 255, blue: 0x16 / 255)
    static let logoSurfaceVariant = Color(red: 0x26 / 255, green: 0x26 / 255, blue: 0x26 / 255)
    static let logoWhite = Color.white
    static let logoGray = Color(red: 0xB3 / 255, green: 0xB3 / 255, blue: 0xB3 / 255)

    static let background = logoBlack
    static let onBackground = logoWhite
    static let surface = logoSurface
    static let onSurface = logoWhite
    static let surfaceVariant = logoSurfaceVariant
    static let onSurfaceVariant = logoGray
}

/// Applica lo stile scuro monocromatico dell'app a tutta la gerarchia di viste,
/// equivalente a `OpticProSuiteTheme` in Compose.
struct AppTheme: ViewModifier {
    func body(content: Content) -> some View {
        content
            .preferredColorScheme(.dark)
            .tint(AppColor.onBackground)
            .background(AppColor.background)
    }
}

extension View {
    func appTheme() -> some View { modifier(AppTheme()) }
}
